### PR TITLE
Add einops as extra dependency

### LINKED ISSUE #185: Error in offline mode with `trust_remote code`: SFR-Embedding-Mistral and nomic does not work without `einops`
### Model description

You have mentioned that sfr-embedding model is supported along with all other huggingface embedding models (ref.nomic).
However, both are not working :
infinity    | ERROR    2024-03-21 14:35:59,554 infinity_emb ERROR:          acceleration.py:21
infinity    |          BetterTransformer is not available for model. The
infinity    |          model type mistral is not yet supported to be used
infinity    |          with BetterTransformer. Feel free to open an issue
infinity    |          at https://github.com/huggingface/optimum/issues if
infinity    |          you would like this model type to be supported.
infinity    |          Currently supported models are: dict_keys(['albert',
infinity    |          'bark', 'bart', 'bert', 'bert-generation',
infinity    |          'blenderbot', 'bloom', 'camembert', 'blip-2',
infinity    |          'clip', 'codegen', 'data2vec-text', 'deit',
infinity    |          'distilbert', 'electra', 'ernie', 'fsmt', 'gpt2',
infinity    |          'gptj', 'gpt_neo', 'gpt_neox', 'hubert', 'layoutlm',
infinity    |          'm2m_100', 'marian', 'markuplm', 'mbart', 'opt',
infinity    |          'pegasus', 'rembert', 'prophetnet', 'roberta',
infinity    |          'roc_bert', 'roformer', 'splinter', 'tapas', 't5',
infinity    |          'vilt', 'vit', 'vit_mae', 'vit_msn', 'wav2vec2',
infinity    |          'xlm-roberta', 'yolos']).. Continue without
infinity    |          bettertransformer modeling code.
infinity    |          Traceback (most recent call last):
infinity    |            File
infinity    |          "/app/infinity_emb/transformer/acceleration.py",
infinity    |          line 19, in to_bettertransformer
infinity    |              model = BetterTransformer.transform(model)
infinity    |            File "/usr/lib/python3.10/contextlib.py", line 79,
infinity    |          in inner
infinity    |              return func(*args, **kwds)
infinity    |            File
infinity    |          "/app/.venv/lib/python3.10/site-packages/optimum/bet
infinity    |          tertransformer/transformation.py", line 234, in
infinity    |          transform
infinity    |              raise NotImplementedError(
infinity    |          NotImplementedError: The model type mistral is not
infinity    |          yet supported to be used with BetterTransformer.

### Open source status

- [ ] The model implementation is available on transformers
- [ ] The model weights are available on huggingface-hub
- [ ] I verified that the model is currently not running in infinity

### Provide useful links for the implementation

_No response_

### LINKED ISSUE #231: Include `einops` in docker image
### Feature request

As the title states, add einops as dependency either for infinity_emb package or in the Dockerfile

### Motivation

Few models rely on `einops` for their modeling code. It would be nice to be able to run these models directly without having to build custom docker image.
example: https://huggingface.co/Snowflake/snowflake-arctic-embed-m-long/blob/08e7a4449e3f07709fb9387bc3172d393a6cc5e2/modeling_hf_nomic_bert.py#L18

I realise that models on HF Hub are free to use arbitrary dependencies but `einops` seems common enough to include for convenience

### Your contribution

Happy to open a PR based on decision

### PR DESCRIPTION
**Pull request by chiragjn**
Closes #231 
Closes #185

### CODE REVIEW

File: libs/infinity_emb/pyproject.toml [hunk 1]
```diff
@@ -28,6 +28,7 @@ transformers = {version = ">4.8.0", optional=true}
 ctranslate2 = {version = "^4.0.0", optional=true}
 optimum = {version = ">=1.16.2", optional=true, extras=["onnxruntime"]}
 hf_transfer = {version=">=0.1.5"}
+einops = {version = "^0.8.0", optional=true}
```
Comment (chiragjn) [inline libs/infinity_emb/pyproject.toml:31]: Should this be a extra group of its own or should it just be part of something like `torch` ?

Comment (michaelfeil) [reply libs/infinity_emb/pyproject.toml:31]: Extra group is good. I would say it might be better to pin with “*” as version and let it auto resolve.

**Review by michaelfeil — APPROVED** [review]
Looks good, with comment above.

**Comment by michaelfeil** [comment]
Lgtm

**Pull request merged**